prusti-common/src/vir/ast/expr.rs

```diff
@@ -80,11 +84,40 @@ pub enum BinOpKind {
     Implies,
 }
 
+
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum FloatConst {
+    FloatConst64(rustc_apfloat::ieee::IeeeFloat<DoubleS>),
+    FloatConst32(rustc_apfloat::ieee::IeeeFloat<SingleS>),
+}
+
+impl PartialEq for FloatConst {
+    fn eq(&self, other: &FloatConst) -> bool {        
+        match (self, other) {
+            (FloatConst::FloatConst64(f1), FloatConst::FloatConst64(f2)) => f1.to_string() == f2.to_string(),
+            (FloatConst::FloatConst32(f1), FloatConst::FloatConst32(f2)) => f1.to_string() == f2.to_string(),
```

Comment:
Why `to_string()` for comparisons?

Reply:
Probably because floats are `PartialEq` only.

Reply:
But this is a `PartialEq` implementation for `FloatConst`. I don't really see how converting to a string helps? If anything, converting to a string could lose information, and I don't expect that the comparison here should do an "equals +/- a small epsilon" kind of comparison.

Reply:
True. I was referring to the `impl Eq for FloatConst {}` below.